src/models/layerMetadata/pycswLayerCatalogRecord.ts

```diff
@@ -230,6 +243,32 @@ export class PycswLayerCatalogRecord extends LayerMetadata implements IPycswCore
     return ret;
   }
 
+  public static getFieldConfigs(): IPropFieldConfigInfo[] {
+    const ret = [];
+    const layer = new PycswLayerCatalogRecord();
+    for (const prop in layer) {
+      const fieldConfigMap = getFieldConfig<PycswLayerCatalogRecord>(layer, prop);
+      if (fieldConfigMap) {
+        const fieldConfig = { prop: prop, ...fieldConfigMap };
+        if (fieldConfigMap.complexType) {
+          fieldConfig.subFields = getFieldConfigClassInfo(fieldConfigMap.complexType.value);
+        }
+
+        ret.push(fieldConfig);
+        // ret.push({
```

Comment:
delete comment or add TODO if it is useful for near future update

Reply:
Done